frontend/src/components/rule-edit/tracking-rule/ArtifactsOwnedParams.vue

```diff
@@ -3,43 +3,50 @@ import { apiGet } from "@/api";
 import type { User } from "@/api/types";
 import { useUserStore } from "@/stores/user";
 import type { QueryFunction } from "react-query/types/core";
-import { ref } from "vue";
+import { useQuery, useQueryClient } from "vue-query";
+import { onUnmounted, ref } from "vue";
 import ArtifactsOwnedSummary from "./ArtifactsOwnedSummary.vue";
 
 const userStore = useUserStore();
 const username = userStore.username;
 const value = ref<string[]>(username ? [username] : []);
 const apiGetUsers = apiGet as QueryFunction<{ users: User[] }>;
-const route = "/users/";
-const getUsers = async (query: string) => {
-  if (!query) {
-    return value.value;
-  }
-  const results = await apiGetUsers({
-    queryKey: [route, { search: query }],
-    meta: undefined,
-  });
-  return results ? results.users.map((u) => u.name) : [];
-};
+const url = "/fasjsonusers/";
+const { isLoading, isError, data, error } = useQuery(url, apiGetUsers);
```

Comment:
As commented above, I don't think that's the right approach, it should search for the partial username instead of loading them all from fasjson, as there will be too many of them.